### PR TITLE
feat(APIv2): RHINENG-8340 implement systems endpoint under reports

### PR DESCRIPTION
**Pull request by skateman**
There are 3 fields that don't work yet and will be addressed separately:
* compliant
* failed_rule_count
* last_scanned

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: spec/integration/v2/systems_spec.rb [hunk 1]
```diff
@@ -294,4 +294,127 @@
...
+        account: user.account,
+        os_major_version: 8,
+        supports_minors: [0],
+        assigned_system_count: 0
+      ).id
+    end
+
+    get 'Request Systems assigned to a Report' do
+      v2_auth_header
+      tags 'Policies'
```
Comment (romanblanco) [inline spec/integration/v2/systems_spec.rb:322]:
```suggestion
      tags 'Report'
```

File: spec/integration/v2/systems_spec.rb [hunk 2]
```diff
@@ -294,4 +294,127 @@
...
+      v2_auth_header
+      tags 'Systems'
+      description 'Returns a System under a Report'
+      operationId 'System'
+      content_types
+
+      parameter name: :system_id, in: :path, type: :string, required: true
+      parameter name: :report_id, in: :path, type: :string, required: true
+
+      response '200', 'Returns a Systemunder a Report' do
```
Comment (romanblanco) [inline spec/integration/v2/systems_spec.rb:401]:
```suggestion
      response '200', 'Returns a System under a Report' do
```

File: spec/integration/v2/systems_spec.rb [hunk 3]
```diff
@@ -294,4 +294,127 @@
...
+        let(:system_id) { item.id }
+        v2_item_schema('system')
+
+        after { |e| autogenerate_examples(e, 'Returns a Systemunder a Report') }
```
Comment (romanblanco) [inline spec/integration/v2/systems_spec.rb:405]:
```suggestion
        after { |e| autogenerate_examples(e, 'Returns a System under a Report') }
```

File: app/models/v2/test_result.rb [hunk 4]
```diff
@@ -10,7 +10,10 @@ class TestResult < ApplicationRecord
     belongs_to :system, class_name: 'V2::System', optional: true
     belongs_to :tailoring, class_name: 'V2::Tailoring'
 
+    has_one :profile, class_name: 'V2::Profile', through: :tailoring
+    has_one :security_guide, class_name: 'V2::SecurityGuide', through: :profile
     has_one :policy, class_name: 'V2::Policy', through: :tailoring
-    has_one :security_guide, class_name: 'V2::SecurityGuide', through: :tailoring
+    has_one :report, class_name: 'V2::Report', through: :policy
```
Comment (romanblanco) [inline app/models/v2/test_result.rb:16]: Why is this not `belongs_to`?

**Review by romanblanco — CHANGES_REQUESTED** [review]
Minor changes needed

File: app/models/v2/test_result.rb (hunk 4 above)
Comment (skateman) [reply app/models/v2/test_result.rb:16]: `belongs_to` requires a field pointing to that record, indirect relations can be only achieved by `through`

**Review by romanblanco — APPROVED** [review]

**Pull request merged**